src/raphael_backend_flask/auth.py

```diff
@@ -1,19 +1,51 @@
-import os
+from sqlite3 import Row
 
 from flask_httpauth import HTTPBasicAuth
 from werkzeug.security import check_password_hash, generate_password_hash
 
+from raphael_backend_flask.db import execute_sql
+
 auth = HTTPBasicAuth()
-users = {
-    user: generate_password_hash(password)
-    for user, password in (
-        item.split(":") for item in os.environ["USERS"].split(",") if item
+
+class User:
+    def __init__(self, username: str, admin: int):
+        self.username: str = username
+        self.roles: list[str] = [ "admin" ] if admin else []
+
+def get_user_sql(username: str) -> Row | None:
+    res = execute_sql(f"SELECT * FROM users WHERE username = ?", (username,))
+    if res:
+        return res[0]
+    return None
+
+def create_user_sql(username: str, password: str, admin: bool) -> None:
+    hashed = generate_password_hash(password)
+    execute_sql(
+        f"INSERT INTO users (username, password_hash, admin) VALUES (?, ?, ?)",
+        (username, hashed, 1 if admin else 0,),
     )
-}
 
+def update_user_password_sql(username: str, password: str) -> None:
+    hashed = generate_password_hash(password)
+    execute_sql(f"UPDATE users SET password_hash = '{hashed}' WHERE username = ?", (username,))
+
+def delete_user_sql(username: str) -> None:
+    execute_sql(f"UPDATE users SET password_hash = ' ' WHERE username = ?", (username,))
```

Comment:
Why set the hash to empty here rather than remove the row?

If it's so that we keep references to the user, I'd probably rather this was called "disable_user_sql" just for clarity, and `password_hash` was set to NULL since that can then be used as an explicit signal that the account is disabled in `verify_pass`.